BDArmory/FX/ShellCasing.cs

```diff
@@ -21,12 +21,10 @@ public class ShellCasing : MonoBehaviour
         void OnEnable()
         {
             startTime = Time.time;
-            Vector3 randV = Random.insideUnitSphere;
-            velocity = initialV + transform.rotation * new Vector3(
-                configV.x + configD * (1f + Mathf.Abs(configV.x)) * randV.x,
-                configV.y + configD * (1f + Mathf.Abs(configV.y)) * randV.y,
-                configV.z + configD * (1f + Mathf.Abs(configV.z)) * randV.z
-            );
+         
+            Vector3 randV = Random.insideUnitSphere * configD;
+            velocity = initialV + transform.rotation *
+                        ((configV.magnitude * randV) + configV);
```

Comment:
This would significantly increase the spread in all the axes, which is far less realistic than the main component of the spread being in the ejection direction (the current implementation).
E.g., this doesn't look good
![Screenshot_20240712_142901](https://github.com/user-attachments/assets/acc10c10-49c6-4e8c-ae26-336a70b9b86a)

Additionally, it requires significantly more computation as `.magnitude` requires computing a `sqrt` which isn't a hardware operation on some architectures and several extra vector operations (which writing it out with `new Vector3(...)` avoids).

Reply:
If the point of this PR is to increase the spread in the transverse direction without significantly affecting the spread in the ejection direction, then something like
```diff
diff --git a/BDArmory/FX/ShellCasing.cs b/BDArmory/FX/ShellCasing.cs
index d4840aa07..cd443d294 100644
--- a/BDArmory/FX/ShellCasing.cs
+++ b/BDArmory/FX/ShellCasing.cs
@@ -25,5 +25,5 @@ void OnEnable()
             velocity = initialV + transform.rotation * new Vector3(
-                configV.x + configD * (1f + Mathf.Abs(configV.x)) * randV.x,
-                configV.y + configD * (1f + Mathf.Abs(configV.y)) * randV.y,
-                configV.z + configD * (1f + Mathf.Abs(configV.z)) * randV.z
+                configV.x + (configD + 0.1f * Mathf.Abs(configV.x)) * randV.x,
+                configV.y + (configD + 0.1f * Mathf.Abs(configV.y)) * randV.y,
+                configV.z + (configD + 0.1f * Mathf.Abs(configV.z)) * randV.z
             );
diff --git a/BDArmory/Weapons/ModuleWeapon.cs b/BDArmory/Weapons/ModuleWeapon.cs
index f86473621..b3b727aab 100644
--- a/BDArmory/Weapons/ModuleWeapon.cs
+++ b/BDArmory/Weapons/ModuleWeapon.cs
@@ -403,3 +403,3 @@ public float GetEngageRange()
         [KSPField]
-        public float shellEjectDeviation = 0.1f;
+        public float shellEjectDeviation = 0.5f;
 

```
would work better.
E.g.,
![Screenshot_20240712_150739](https://github.com/user-attachments/assets/6af38798-8829-4a8e-a0e9-587cbb89770a)
![Screenshot_20240712_150803](https://github.com/user-attachments/assets/4404946e-96a1-4523-a5bb-8bf5d7ea77b4)

Reply:
I'm gonna assume the way you did it will meet the requirements for controlled velocity, spread, and direction and close the PR

Reply:
I haven't added the changes in yet, but essentially, what I wrote above is giving the shells a spread based on 10% of the ejection velocity (the `0.1f * Mathf.Abs(configV.xyz)` part) plus a configurable spread in all directions (the `configD` part), making the spread shaped like an ellipsoid. It keeps the default fairly similar to how it originally was for guns ejecting shells out the side while allowing for more customisation.
To be fully customisable, it'd need a second deviation config value for the `0.1f` factor (for controlling the length of the major axis of the ellipsoid), but I think that this should be sufficient for most guns.
I can commit those changes if that's sufficient for what you want, otherwise, we can add in the second deviation config (maybe by making `configD` a `Vector2`)?